crypto/ake/mcnoise/src/symmetric_state.rs

```diff
@@ -138,8 +138,8 @@ where
     ///
     /// Updates our hash to reflect a new hash of itself and the provided
     /// data.
-    pub fn mix_hash(&mut self, data: &[u8]) {
-        self.hash += data;
+    pub fn mix_hash<D: AsRef<[u8]>>(&mut self, data: D) {
```

Comment:
this very small change allows to avoid writing `.as_ref()` in a lot of places

Reply:
It could in theory, but I think it doesn't as long as the thing you are passing is actually a ref. But at this specific revision, yes it is a copy, because the `get_bytes()` thing in this revision is basically the same as `to_bytes` in the `ReprBytes` PR.

I'm going to rebase this branch on the `ReprBytes` PR and get rid of the `GetBytes` thing so we can see better what the end state is actually going to be. It will make the PR larger but I think this PR is not that big in the first place. I guess I'm going to make that a third PR and then close this one and the other one?

Reply:
i guess we should revert this in the next version, since you're right, it could be a performance leak